docs/migration/cosmoshub-4-vega-upgrade.md

```diff
@@ -1,4 +1,217 @@
+
+### method I: manual upgrade
+Run Gaia v5.0.x till upgrade height, the node will panic:
+```shell
+ERR UPGRADE "Vega" NEEDED at height: 8695000
+
+panic: UPGRADE "Vega" NEEDED at height: 8695000
+```
+Stop the node, and install Gaia v6.0.0 and re-start by `gaiad start`.
+
+It may take 20 min to a few hours till validators with total sum power > 2/3 complete their nodes upgrades and by that moment, the chain can continue to produce blocks.
+
+### Method II: upgrade by cosmovisor with manually prepare the Gaia v6.0.0 binary
+#### Preparation
+
+Install cosmovisor latest version:
+```shell
+go install github.com/cosmos/cosmos-sdk/cosmovisor/cmd/cosmovisor@latest
+```
+
+Create a cosmovisor folder:
+
+create cosmovisor folder inside gaia home and move Gaia v5.0.0 into ` $GAIA_HOME/cosmovisor/genesis/bin`
+```shell
+mkdir -p $GAIA_HOME/cosmovisor/genesis/bin
+cp $(which gaiad) $GAIA_HOME/cosmovisor/genesis/bin
+```
+build Gaia v6.0.0, and move gaiad v6.0.0 to `$GAIA_HOME/cosmovisor/upgrades/Vega/bin`
+
+```shell
+mkdir -p  $GAIA_HOME/cosmovisor/upgrades/Vega/bin
+cp $(which gaiad) $GAIA_HOME/cosmovisor/upgrades/Vega/bin
+```
+Then you should get the following structure:
+```shell
+.
+├── current -> genesis or upgrades/<name>
+├── genesis
+│   └── bin
+│       └── gaiadv5.0.0
+└── upgrades
+└── Vega
+└── bin
+    └── gaiadv6.0.0
+```
+Export the environmental variables:
+```shell
+export DAEMON_NAME=gaiad
+# please change to your own gaia home dir
+export DAEMON_HOME= $GAIA_HOME
+export DAEMON_RESTART_AFTER_UPGRADE=true
+```
+
+Start the node: 
+```shell
+cosmovisor start --x-crisis-skip-assert-invariants
+```
+ ????
+ 
+#### Expected ugprade result
+When the upgrade block height is reached, you can find the following information in the log: 
+```shell
+ERR UPGRADE "Vega" NEEDED at height: 8695000: upgrade to Vega and applying upgrade "Vega" at height:8695000.
+```
+ This may take 20 min to a few hours.
+Then the chain will continue to produce blocks after validators with total sum power > 2/3 complete their nodes upgrades.
+
+### Method III: upgrade by cosmovisor with manually prepare the Gaia v6.0.0 binary (not recommended!!!)
+#### Preparation
+Install cosmovisor v0.1
+```shell
+go install github.com/cosmos/cosmos-sdk/cosmovisor/cmd/cosmovisor@v0.1.0
+
+```
+Create a cosmovisor folder:
+
+create a cosmovisor folder inside gaia home and move gaiad v5.0.0 into ` $GAIA_HOME/cosmovisor/genesis/bin`
+```shell
+mkdir -p $GAIA_HOME/cosmovisor/genesis/bin
+cp $(which gaiad) $GAIA_HOME/cosmovisor/genesis/bin
+```
+
+```shell
+.
+├── current -> genesis or upgrades/<name>
+└── genesis
+     └── bin
+        └── gaiadv5.0.0
+```
+Export the environmental variables:
+```shell
+export DAEMON_NAME=gaiad
+# please change to your own gaia home dir
+export DAEMON_HOME= $GAIA_HOME
+export DAEMON_RESTART_AFTER_UPGRADE=true
+export DAEMON_ALLOW_DOWNLOAD_BINARIES=true
+```
+Start the node:
+```shell
+cosmovisor start --x-crisis-skip-assert-invariants
+```
+????
```

Comment:
Isn't this exactly the same as method II? Nothing here talks about the auto-download function?